source-mysql/main.go

```diff
@@ -245,22 +258,57 @@ func (db *mysqlDatabase) connect(ctx context.Context) error {
 	return nil
 }
 
-func (db *mysqlDatabase) getBinlogExpiry() (time.Duration, error) {
+var timeZoneOffsetRegex = regexp.MustCompile(`^[-+][0-9]{1,2}:[0-9]{2}$`)
+
+func queryTimeZone(conn *client.Conn) (*time.Location, error) {
+	var tzName, err = queryStringVariable(conn, `SELECT @@GLOBAL.time_zone;`)
+	if err != nil {
+		return nil, fmt.Errorf("error querying 'time_zone' system variable: %w", err)
+	}
+	logrus.WithField("time_zone", tzName).Debug("queried time_zone system variable")
+	if tzName == "SYSTEM" {
+		return nil, errDatabaseTimezoneUnknown
+	}
+
+	// If it looks like a numeric offset then construct a fixed-offset time zone from that.
+	if timeZoneOffsetRegex.MatchString(tzName) {
```

Comment:
Should we just ... not have this behavior? It's extra code to maintain that could have subtle bugs, it's more documentation to write and to communicate to users, and it doesn't seem unreasonable to simply require an IANA time zone name.

If we do keep it, it looks like it needs test coverage (only IANA case is covered currently)

Reply:
I included this logic because it seemed reasonable to support any legal setting of the `time_zone` variable that wasn't `'SYSTEM'`.

I'm not super attached to keeping it, but it feels worthwhile to me to support as many customer database configurations as we reasonably can, and parsing the numeric offset isn't super complicated (and in fact now that I'm well rested I've realized we can get Go to do the parsing with something like `t, _ := time.Parse("-07:00", tzName); t.Location()` so that avoids hand-written parsing code here).

So I'm inclined to keep this support but replace the hand-written parser with Go's `time.Parse` and then add a new test case which exercises DATETIME support in more time zones than just `America/Chicago`.